### ISSUE TITLE
[RFC] [R-package] decouple verbosity and validation sets in lightgbm()

### ISSUE THREAD

**Issue by jameslamb** [microsoft/LightGBM#5203]
## Summary

`lightgbm::lightgbm()` in the R package currently has the behavior "if logging verbosity is set higher than 0, add the training data to the validation sets used to compute eval metrics and for early stopping".

This behavior should be removed.

## Motivation

I think this coupling of "what level of logs do I want?" and "what validation sets do I want to use?" is confusing and surprising.

## Description

To do this, I think the following condition should be removed:

https://github.com/microsoft/LightGBM/blob/14ca8fc07e3411fac215c4009ed47aa019fad980/R-package/R/lightgbm.R#L196-L199

And `valids` should be promoted to a top-level keyword argument.

This is a breaking change, and if it's done should be done prior to the v4.0.0 release (#5153).

## References

Created based on https://github.com/microsoft/LightGBM/pull/5199#discussion_r868583048.

**Comment by jmoralez** [comment]
I'm a bit confused about the role of the `lightgbm::lightgbm` function. The description says `Simple interface for training a LightGBM model.` and it has some user friendly arguments like `data` and `label` but also has `early_stopping_rounds` which seems like a more advanced feature as well as passing `valids` through `...`.

I support removing the behavior of the `verbosity > 0` meaning: "I want to use the training set to log metrics and for early stopping" because if this interface is meant for beginners they might think that their model is very good when it really isn't. Promoting `valids` to a top level argument makes this function more like `lgb.train` so I think we should maybe consider making this one simpler instead?

**Comment by jameslamb** [comment]
> I'm a bit confused about the role of the `lightgbm::lightgbm()` function. The description says `Simple interface for training a LightGBM model.`

Thanks for raising this concern. Let's separate "remove this unexpected behavior" and "maybe promote `valids` to a top-level keyword argument". It's already possible to pass `valids`  into `lightgbm::lightgbm()` via `...` so that other decision isn't urgent.

I think that "simple" was the original intent when the R package was first introduced in #168, but that the difference between `lightgbm()` and `lgb.train()` might not have been carefully considered.

The R package seems to have mostly been copied from `{xgboost}`, and you can see that even today `{xgboost}` uses the same language.

https://github.com/dmlc/xgboost/blob/94ca52b7b7801e709cf68aa06de024b361df577b/R-package/R/xgboost.R#L1

Regardless of the history, I think right now as we prepare for `v4.0.0`, we should more clearly define the relationship between these two functions. Instead of the dividing line being about whether one is "simple" and another is "advanced", I'd like to make more explicit the way that I personally have been approaching these functions:

* `lgb.train()` = low-level interface to LightGBM, focused on performance
* `lightgbm()` = high-level interface to LightGBM, focused on compatibility with the rest of the R modeling ecosystem (i.e., intended to be be the more idiomatic, "R-thonic" interface)
